src/js/components/Form/Form.js

```diff
@@ -64,14 +64,55 @@ const Form = forwardRef(
       defaultValidationResults,
     );
 
-    useEffect(
-      () => setValidationResults({ errors: errorsProp, infos: infosProp }),
-      [errorsProp, infosProp],
-    );
+    // when onBlur input validation is triggered by a click event,
+    // we need to complete the click event before running the validation
+    // otherwise, click events like reset, etc. may not be registered.
+    // for a detailed scenario/discussion, see: https://github.com/grommet/grommet/issues/4863
+    const [pendingValidation, setPendingValidation] = useState(undefined);
+
+    useEffect(() => {
+      setPendingValidation(undefined);
+      setValidationResults({ errors: errorsProp, infos: infosProp });
+    }, [errorsProp, infosProp]);
     const validations = useRef({});
 
+    useEffect(() => {
+      const timer = setTimeout(() => {
+        if (pendingValidation) {
+          // run validations on touched keys
+          const [nextErrors, nextInfos] = validate(
+            Object.entries(validations.current).filter(
+              ([n]) => touched[n] || n === pendingValidation,
+            ),
+            value,
+          );
+          setPendingValidation(undefined);
+          // give user access to errors that have occurred on validation
+          setValidationResults(prevValidationResults => {
+            // keep any previous errors and infos for untouched keys,
+            // which probably came from a submit
+            const nextValidationResults = {
+              errors: { ...prevValidationResults.errors, ...nextErrors },
+              infos: { ...prevValidationResults.infos, ...nextInfos },
+            };
+            if (onValidate) onValidate(nextValidationResults);
+            return nextValidationResults;
+          });
+        }
+        // a timeout is needed to ensure that a click event (like one on a reset
+        // button) completes prior to running the validation. without a timeout,
+        // the blur will always complete and trigger a validation prematurely
+        // The following values have been empirically tested, but 120 was
+        // selected because it is the largest value
+        // Chrome: 100, Safari: 120, Firefox: 80
+      }, 120);
```

Comment:
comment from shimi: we might want to make this value themeable, so if callers still experience the issue it would be configurable to manipulate the timeout value to make the onBlur trigger faster or slower.